### PR TITLE
chore: move throttle to duration

### PR DESCRIPTION
**Pull request by peternhale**
@W-14867258@

Create a Duration class that applies throttle factor

### CODE REVIEW

File: test/specs/anInitialSuite.e2e.ts [hunk 1]
```diff
@@ -96,40 +95,40 @@ describe('An Initial Suite', async () => {
...
+    await expect(commands).toContain('SFDX: Create Apex Trigger');
+    // There are more, but just look for the first few commands.
+
+    // Escape out of the pick list.
+    await prompt.cancel();
+  });
 
   after('Tear down and clean up the testing environment', async () => {
-    await testSetup.tearDown();
+    await testSetup?.tearDown();
```
Comment (daphne-sfdc) [inline test/specs/anInitialSuite.e2e.ts:132]: Is this safe navigation operator a change we would like to make to all E2E tests or just AnInitialSuite as a special case?

Comment (peternhale) [reply test/specs/anInitialSuite.e2e.ts:132]: It might be worthwhile, but no rush.

File: test/utilities/cliCommands.ts [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    throw new Error('cannot setup lwc tests without a project folder.');
+  }
+
+  const jestInstallResult = await runCliCommand('force:lightning:lwc:test:setup', {
+    cwd: projectFolder
+  });
+
+  if (jestInstallResult.exitCode) {
+    log(
+      `alias failed. Exit code: ${jestInstallResult.exitCode}. \nRaw stderr: ${jestInstallResult.stderr}`
```
Comment (daphne-sfdc) [inline test/utilities/cliCommands.ts:182]:
```suggestion
      `setup lwc tests failed. Exit code: ${jestInstallResult.exitCode}. \nRaw stderr: ${jestInstallResult.stderr}`
```

File: test/utilities/cliCommands.ts [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+  const sfOrgCreateUserResult = await runCliCommand(
+    'org:create:user',
+    '--definition-file',
+    systemAdminUserDefPath,
+    '--target-org',
+    targetOrg
+  );
+  if (sfOrgCreateUserResult.exitCode) {
+    log(
+      `org crate user failed Exit code: ${sfOrgCreateUserResult.exitCode}. \nRaw stderr: ${sfOrgCreateUserResult.stderr}`
```
Comment (daphne-sfdc) [inline test/utilities/cliCommands.ts:205]:
```suggestion
      `org create user failed Exit code: ${sfOrgCreateUserResult.exitCode}. \nRaw stderr: ${sfOrgCreateUserResult.stderr}`
```

File: test/utilities/cliCommands.ts [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    );
+    throw new Error(sfOrgCreateUserResult.stderr);
+  }
+  debug(`createUser results ${JSON.stringify(sfOrgCreateUserResult)}`);
+  return sfOrgCreateUserResult;
+}
+
+export function removedEscapedCharacters(result: string): string {
```
Comment (daphne-sfdc) [inline test/utilities/cliCommands.ts:213]: nit - change function name to present tense

```suggestion
export function removeEscapedCharacters(result: string): string {
```

File: README.md [hunk 5]
```diff
@@ -51,6 +51,9 @@ The following is a list of environment variables that are used with this project
 - THROTTLE_FACTOR
   - Default value: `1`
 
+- SFDX_AUTH_URL
+  - Provides the dev hub auth URL to be used to authenticate the dev hub as part of test setup. There is no default value.
+
```
Comment (daphne-sfdc) [inline README.md:56]: Thanks for the documentation update! 😀

**Review by daphne-sfdc — APPROVED** [review]
Approved! 🎉 

Thanks @peternhale for the great refactoring and improvements!

Passing E2E test runs: ✅ 

- Apex and Core: https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/9998746770
- Everything else: https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/9998750143

**Pull request merged**